lib/web_ui/lib/src/engine/html_image_codec.dart

```diff
@@ -204,6 +200,52 @@ class HtmlImage implements ui.Image {
     }
   }
 
+  /// Converts [imgElement] to straight RGBA.
+  /// Prefer [WebGLContext] whenever possible since Canvas keeps the RGBA as premultiplied-alpha
+  /// and [DomCanvasRenderingContext2DExtension.getImageData] converts back to straight RGBA
+  /// which is lossy algorithm for instance when Alpha chanel is 0
+  /// then there is no way to get back original straight RGBA
+  /// https://stackoverflow.com/questions/23497925/how-can-i-stop-the-alpha-premultiplication-with-canvas-imagedata
+  ByteBuffer _getStraightRgba() {
```

Comment:
(same as in https://github.com/flutter/engine/pull/39109#discussion_r1195766631)

After reading the image data back, set `canvas.width` and `canvas.height` to zero. Safari on iOS will crash if the GC doesn't collect 2D canvases fast enough. Setting the image size to zero causes Safari to free internal image memory immediately.